File: src/main/java/com/puhovin/intellijplugin/twm/model/SettingsMode.java
package com.puhovin.intellijplugin.twm.model;

import java.io.Serializable;

public enum SettingsMode implements Serializable {
    GLOBAL(true),
    PROJECT(false);


    private final boolean value;

    SettingsMode(boolean value) {
        this.value = value;
    }

    public boolean getValue() {
        return value;
    }

    public static SettingsMode fromBoolean(boolean value) {
        return value ? GLOBAL : PROJECT;
    }
}

File: src/main/kotlin/com/puhovin/intellijplugin/twm/model/SettingsMode.kt
package com.puhovin.intellijplugin.twm.model

import java.io.Serializable

enum class SettingsMode(val value: Boolean) : Serializable {
    GLOBAL(true),
    PROJECT(false);

    companion object {
        fun fromBoolean(value: Boolean) = if (value) GLOBAL else PROJECT
    }
}